File: Algoritmo viajero/viajero.cpp
#include <iostream>
#include <string.h>
#include <list>
#include <ctype.h>
#include "funciones.h"

void Permutaciones(string cad, int l=0);

int main() {
	int nodos,conexiones, costo, mascorto=2147483647;
	char v1, v2;
	int matriz[10][10];
	cin >> nodos >> conexiones;
	string palabra = "", camino, caminoCorto = "";
	inicializarMatriz(matriz,nodos);
	for (int j = 0; j < conexiones; ++j)
	{
		cin >> v1 >> v2 >> costo;
		matriz[(int) (toupper(v1)-65)][(int) (toupper(v2)-65)] = costo;
		matriz[(int) (toupper(v2)-65)][(int) (toupper(v1)-65)] = costo;
	}
	for (int i = 0; i < nodos; ++i)
		palabra += (char) (i+65);
	Permutaciones(palabra);
	int tam = caminos.size();
	for (int i = 0; i < tam; ++i)
	{
		camino = caminos.front();
		caminos.pop_front();
		cout << "Camino " << i+1 << "\t"; 
		imprimeCamino(camino, matriz, mascorto, caminoCorto);
	}
	cout << "El camino más corto es: " << caminoCorto <<  mascorto << endl;
	cin.get();
	return 0;
}

File: Ruby/Extra/Viajero/viajero.rb
#!/usr/bin/env ruby
#coding: UTF-8

nodos = gets.to_i
arcos = gets.to_i
matriz = Array.new(nodos) {Array.new(nodos) {0}} #Matriz llena de ceros
permutaciones = []
arcos.times do
	x, y, valor = gets.split
	permutaciones.push x.upcase, y.upcase
	x = (x.upcase.ord - 65)
	y = (y.upcase.ord - 65)
	valor = valor.to_i
	matriz[x][y] = valor
	matriz[y][x] = valor
end
#Para crear una matriz con las permutaciones
permutaciones =  permutaciones.uniq!.permutation.to_a
permutaciones.each {|i| i.push i[0]}
mascorto = 2e10
indice = 0
for i in permutaciones
	cad = "El camino #{indice} \t"
	costo = 0
	b = true
	for j in 0...(i.length-1)
		x = i[j].ord - 65 
		y = (i[j+1]).ord - 65
		cad += (i[j] + ", ")
		costo += matriz[x][y]
		b = false if matriz[x][y] == 0
	end
	cad += i.last + " " + costo.to_s
	puts cad 
	mascorto, $camino_corto = costo, cad if (mascorto > costo and b)
	indice += 1
end
puts $camino_corto + " es el más corto"